File: Практическая работа №3/src/main/java/io/github/ikbo0619/testing/team6/durak/Player.java
package io.github.ikbo0619.testing.team6.durak;

import java.util.ArrayList;

public class Player {

    private Hand hand;
    private String name;
    private final int id = 0;
    private Deck deck;
    private static int count = 0;
    private boolean attacker;


    public Player(Deck d, String n) {
    }

    public void draw(Deck d) {
    }

    public void draw() {
    }

    public void drawCards(int n) {
    }

    public void discard(Card c) {
    }

    public int cardsInHand() {
        return 0;
    }

    // Проверка победы
    public boolean victoryAchieved() {
        return false;
    }

    public boolean isAttacker() {
        return false;
    }

    public void makeAttacker() {
    }

    public void switchRole() {
    }

    @Override
    public String toString() {
        return "";
    }

    public Card getCard(int i) {
        return null;
    }

}

File: Практическая работа №6/№3 Имеющаяся программная система/src/main/kotlin/Player.kt
class Player(private val deck: Deck, private val name: String) {
    private val hand = Hand()
    var attacker = false

    init {
        drawCards(6)
    }

    private fun draw(deck: Deck) {
        if (!deck.isEmpty())
            hand.add(deck.draw()!!)
    }

    private fun drawCards(number: Int) {
        for (i in 0 until number)
            draw()
    }

    private fun draw() = draw(deck)

    fun takeCard(card: Card) = hand.add(card)

    fun cardsInHand() = hand.size()

    fun replenish() {
        val toDraw = hand.numberToDraw()
        drawCards(toDraw)
    }

    fun victoryAchieved() = hand.size() <= 0 && deck.isEmpty()

    fun switchRole() {
        attacker = !attacker
    }

    override fun toString() = "'$name'"

    fun getCard(index: Int) = hand.getCardByIndex(index - 1)

    fun useCard(index: Int) = hand.useCardByIndex(index - 1)

    fun cardList() = buildString {
        append("\n=== YOUR CARDS ===\n\n")

        hand.cards.forEachIndexed { index, card ->
            append("${index + 1} ~ $card\n")
        }

        append("\n")
    }
}
